mingw crossdist builds - requirements

Is there any documentation what is needed for mingw-w64 builds?

Only installing mingw-w64 seems not to be enough. Wine is also needed (but is not checked in cmake). Afterwards I still see the following in build outputs:
```
it looks like wine32 is missing, you should install it.
multiarch needs to be enabled first.  as root, please
execute "dpkg --add-architecture i386 && apt-get update &&
apt-get install wine32"
```

builds without wine fail with:
```
[ 10%] Building CXX object src/tools/kdb/CMakeFiles/kdb-objects.dir/metaget.cpp.obj
Scanning dependencies of target kdberrors_generated
[ 11%] Generating ../include/kdberrors.h
/bin/sh: 1: elektra-export-errors_EXE_LOC-NOTFOUND: not found
src/error/CMakeFiles/kdberrors_generated.dir/build.make:61: recipe for target 'src/include/kdberrors.h' failed
make[2]: *** [src/include/kdberrors.h] Error 127
CMakeFiles/Makefile2:281: recipe for target 'src/error/CMakeFiles/kdberrors_generated.dir/all' failed
make[1]: *** [src/error/CMakeFiles/kdberrors_generated.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

which is because https://github.com/ElektraInitiative/libelektra/blob/36f2a2a0fd8b961ad5989a8a8a1fa05006a8ba2a/cmake/Modules/LibAddMacros.cmake#L164 does not check if wine is actually found.

--- a/cmake/ElektraCompiling.cmake
+++ b/cmake/ElektraCompiling.cmake
@@ -64,6 +64,14 @@ if (CMAKE_COMPILER_IS_GNUCXX)
 	execute_process (COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
 	if (WIN32)
 		message (STATUS "mingw detected")
+
+		# mingw builds need wine to be installed
+		find_program (WINE "wine")
+		if (WINE)
+			message (STATUS "wine detected")
+		else (WINE)
+			message (FATAL_ERROR "wine could not be found but is needed for mingw builds")
+		endif (WINE)
 	else (WIN32)
 
 		# not supported by icc:

--- a/doc/BUILDSERVER.md
+++ b/doc/BUILDSERVER.md
@@ -101,7 +101,6 @@ build:
 * jenkins build [icc](https://build.libelektra.org/job/elektra-icc/) please
 * jenkins build [libelektra](https://build.libelektra.org/jenkins/job/libelektra/) please
 * jenkins build [local-installation](https://build.libelektra.org/job/elektra-local-installation/) please
-* jenkins build [mingw64](https://build.libelektra.org/job/elektra-gcc-configure-mingw-w64/) please
 * jenkins build [source-package-test](https://build.libelektra.org/job/elektra-source-package-test/) please
 
 Additionally `jenkins build all please` can be used to trigger all build jobs

--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@ -201,6 +201,7 @@ These notes are of interest for people developing Elektra:
    [Google Test][] instead, please set the value of `GTEST_ROOT` to the path of you local copy of the
    [Google Test][] framework. *(René Schwaiger)*
 - [Google Test][] is installed in Docker images used by the build system. *(Lukas Winkler)*
+- Port `elektra-gcc-configure-mingw-w64` to new build system. *(Lukas Winkler)*
 - Port `debian-multiconfig-gcc-stable` to new build system. *(Lukas Winkler)*
 - Port `elektra-ini-mergerequests` to new build system. *(Lukas Winkler)*
 

--- a/scripts/docker/debian/stretch/Dockerfile
+++ b/scripts/docker/debian/stretch/Dockerfile
@@ -56,6 +56,7 @@ RUN apt-get update \
         libgirepository1.0-dev \
         systemd \
         libsystemd-dev \
+        mingw-w64 \
         llvm \
         icheck \
         devscripts \

--- a/scripts/docker/debian/stretch/Dockerfile
+++ b/scripts/docker/debian/stretch/Dockerfile
@@ -57,6 +57,7 @@ RUN apt-get update \
         systemd \
         libsystemd-dev \
         mingw-w64 \
+        wine \
         llvm \
         icheck \
         devscripts \

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -325,6 +325,7 @@ def generateFullBuildStages() {
             CMAKE_FLAGS_CLANG,
         [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
+    tasks << buildAndTestMingwW64(),
     tasks << buildAndTest(
         "ubuntu-xenial",
         DOCKER_IMAGES.xenial,
@@ -503,6 +504,28 @@ def buildAndTestAsan(testName, image, extraCmakeFlags = [:]) {
                              "ASAN_SYMBOLIZER_PATH=${llvm_symbolizer}"]){
                         ctest()
                     }
+                }
+            }
+        }
+    }]
+}
+
+/* Helper to generate mingw test */
+def buildAndTestMingwW64() {
+    def testName = "debian-stable-mingw-w64"
+    return [(testName): {
+        stage(testName) {
+            withDockerEnv(DOCKER_IMAGES.stretch) {
+                dir('build') {
+                    deleteDir()
+                    sh '../scripts/configure-mingw-w64 ..'
+                    sh 'make'
+                    def destdir='elektra'
+                    withEnv(["DESTDIR='${destdir}'"]){
+                        sh 'make install'
+                    }
+                    sh "zip -r elektra.zip ${destdir}"
+                    archive(['elektra.zip'])
                 }
             }
         }

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -325,7 +325,7 @@ def generateFullBuildStages() {
             CMAKE_FLAGS_CLANG,
         [TEST.ALL, TEST.MEM, TEST.INSTALL]
     )
-    tasks << buildAndTestMingwW64(),
+    tasks << buildAndTestMingwW64()
     tasks << buildAndTest(
         "ubuntu-xenial",
         DOCKER_IMAGES.xenial,

--- a/scripts/jenkins/Jenkinsfile
+++ b/scripts/jenkins/Jenkinsfile
@@ -521,7 +521,7 @@ def buildAndTestMingwW64() {
                     sh '../scripts/configure-mingw-w64 ..'
                     sh 'make'
                     def destdir='elektra'
-                    withEnv(["DESTDIR='${destdir}'"]){
+                    withEnv(["DESTDIR=${destdir}"]){
                         sh 'make install'
                     }
                     sh "zip -r elektra.zip ${destdir}"

